File: src/TCC.Contabilidade.Domain/Entities/Convite.cs
namespace TCC.Contabilidade.Domain.Entities;

public class Convite
{
    public Guid Id { get; set; }

    public string EmailCliente { get; set; } = string.Empty;

    public string Token { get; set; } = string.Empty;

    public bool Utilizado { get; set; }

    public Guid ContadorId { get; set; }

    public DateTime Expiracao { get; set; }

    public bool Usado { get; set; }
}

File: src/TCC.Contabilidade.Web/src/interfaces/convite.ts
export interface Convite {
  token: string;
  emailDestino: string;
  dataExpiracao: string;
  utilizado: boolean;
}

export interface CriarConviteRequest {
  emailCliente: string;
}

export interface CriarConviteResponse {
  mensagem: string;
  token: string;
}
